File: Contiguous.java
/*
  Time Complexity : O(n))
  Space Complexity : O(n)

 Did this code successfully run on Leetcode : YES
 Any problem you faced while coding this :No 

Approach:Here we are using the running sum and hashmp 
 */
class
class Solution {
    public int findMaxLength(int[] nums) {
        if(nums == null || nums.length<=0){
            return 0;
        }
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        map.put(0,-1);
        int max = 0;
        int sum =0;
        for(int i =0 ;i< nums.length;i++){
            if(nums[i] == 0){
                sum--;
            }if(nums[i] == 1){
                sum++;
            }
            if(map.containsKey(sum)){
                max = Math.max(max, i - map.get(sum));
            }else{
                map.put(sum,i);
            }
        }
        return max;
    }
}

File: Contiguous.py
// Time Complexity : o(n)
// Space Complexity : o(n)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this : no
class Solution(object):
    def findMaxLength(self, nums):
        diff=0
        count=0
        d={}
        d[0]=-1
        for i in range(len(nums)):
            if nums[i]==1:
                diff+=1
            else:
                diff-=1
            if diff not in d:
                d[diff]=i
            else:
                count=max(count,i-d[diff])
        return count
                
